### PR TITLE
Refactored bridge and created tests

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/FastRoute.php [hunk 1]
```diff
@@ -212,37 +214,29 @@ public function match(Request $request): MatchingResult
...
-        if (!empty($parameters['defaults'])) {
-            $defaultValues = $parameters['defaults'];
-        }
-
-        $routeParser = new RouteParser();
-        $routes = array_reverse($routeParser->parse($route->getPattern()));
+        /**
+         * TODO may loop be removed?
+         * what's it for?
+         * more than one route by name cannot exist
```
Comment (samdark) [inline src/FastRoute.php]: Do you mean `foreach ($routes as $parts) {`?

File: src/FastRouteFactory.php [hunk 2]
```diff
@@ -5,24 +5,24 @@
...
 use FastRoute\RouteCollector;
 use FastRoute\RouteParser\Std;
-use Psr\Container\ContainerInterface;
 
 class FastRouteFactory
 {
-    public function __invoke(ContainerInterface $container)
+    public function __invoke()
     {
+        // TODO may it be used via di?
```
Comment (samdark) [inline src/FastRouteFactory.php]: I don't think there's a need for it. That is fairly concrete adapter.

File: src/RouteNotFoundException.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Yiisoft\Router\FastRoute;
+
+use Throwable;
+
+// TODO maybe it should be in yiisoft/router?
```
Comment (samdark) [inline src/RouteNotFoundException.php]: Yes. Definitely.

**Comment by samdark** [comment]
Overall great direction 👍

File: src/FastRouteFactory.php (hunk 2 above)
Comment (xepozz) [reply src/FastRouteFactory.php]: Ok

File: src/FastRoute.php (hunk 1 above)
Comment (xepozz) [reply src/FastRoute.php]: Yes, sure. $parsedRoutes always contains one element.

Comment (samdark) [reply src/FastRoute.php]: Yes. Likely could be remove then.

File: src/RouteNotFoundException.php (hunk 3 above)
Comment (xepozz) [reply src/RouteNotFoundException.php]: Moved.
https://github.com/yiisoft/router/pull/12

File: src/FastRouteFactory.php (hunk 2 above)
Comment (xepozz) [reply src/FastRouteFactory.php]: Deleted

File: src/FastRoute.php (hunk 1 above)
Comment (xepozz) [reply src/FastRoute.php]: Removed.

File: src/FastRoute.php [hunk 4]
```diff
@@ -112,6 +111,10 @@ class FastRoute implements RouterInterface
      * @var Route[]
      */
     private $routesToInject = [];
+    /**
+     * @var \FastRoute\RouteParser\Std
+     */
+    private $routerParser;
```
Comment (samdark) [inline src/FastRoute.php]:
```suggestion
    private $routeParser;
```

File: src/FastRoute.php [hunk 5]
```diff
@@ -127,22 +130,20 @@ class FastRoute implements RouterInterface
      *
      * @param null|RouteCollector $router If not provided, a default
      *     implementation will be used.
+     * @param \FastRoute\RouteParser\Std $routerParser
```
Comment (samdark) [inline src/FastRoute.php]:
```suggestion
     * @param \FastRoute\RouteParser\Std $routeParser
```

File: src/FastRoute.php [hunk 6]
```diff
@@ -127,22 +130,20 @@ class FastRoute implements RouterInterface
...
      * @param null|callable $dispatcherFactory Callable that will return a
      *     FastRoute dispatcher.
      * @param array $config Array of custom configuration options.
      */
     public function __construct(
-        RouteCollector $router = null,
+        RouteCollector $router,
+        Std $routerParser,
```
Comment (samdark) [inline src/FastRoute.php]:
```suggestion
        Std $routeParser,
```

**Comment by samdark** [comment]
Thanks!

**Pull request merged**